storage/s3.go

```diff
@@ -89,22 +85,48 @@ func parseEndpoint(endpoint string) (urlpkg.URL, error) {
 }
 
 // NewS3Storage creates new S3 session.
-func NewS3Storage(opts S3Options) (*S3, error) {
+func NewS3Storage(opts StorageOptions) (*S3, error) {
 	endpointURL, err := parseEndpoint(opts.Endpoint)
 	if err != nil {
 		return nil, err
 	}
 
-	awsSession, err := newSession(opts)
+	awsSession, err := newSession(S3Options{
+		Region:      opts.DestinationRegion,
+		Endpoint:    opts.Endpoint,
+		MaxRetries:  opts.MaxRetries,
+		NoVerifySSL: opts.NoVerifySSL,
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	return &S3{
+	s3Storage := &S3{
 		api:         s3.New(awsSession),
 		downloader:  s3manager.NewDownloader(awsSession),
 		uploader:    s3manager.NewUploader(awsSession),
 		endpointURL: endpointURL,
+	}
+	if opts.SourceRegion == "" || opts.SourceRegion == aws.StringValue(awsSession.Config.Region) {
+		s3Storage.destinationS3 = s3Storage
+		return s3Storage, nil
+	}
+
+	dstSession, err := newSession(S3Options{
```

Comment:
 As I wrote in the above comment, I could not understand why we need to create two sessions with one `NewS3Storage()` call.

Reply:
so, if multiple commands are entered using batch mode, for each copy/move operation we would need to know source and destination clients